src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs

```diff
@@ -537,6 +541,17 @@ public bool TryCreateStringContent(int index, string columnName, bool singleColu
 
                 #endregion
             }
+
+            private static string GetDiagnosticUpdatedMessage(DiagnosticsUpdatedArgs e)
+            {
+                var id = e.Id.ToString();
+                if (e.Id is AnalyzerUpdateArgsId analyzer)
+                {
+                    id = analyzer.Analyzer.ToString();
+                }
+
+                return $"{e.Workspace.Kind} {id} {e.Kind} {(object)e.DocumentId ?? e.ProjectId} {e.Diagnostics.Length}";
```

Comment:
How often does this get called - do the allocations here matter?  @sharwell?

Reply:
without option on, this never get called. it uses our Logger framework, which only build string to log when logging option is on.

Reply:
Got it, thanks.